### PR TITLE
Hardcoded secrets test [APPSEC-11807]

### PR DESCRIPTION
**Pull request by iunanua**
## Description

Includes a nodejs test for the new `HARDCODED_SECRET` vulnerability.

In the case of nodejs, the vulnerability is detected in the application startup so it differs a bit from the rest of the existing sink tests.

`test_insecure` method checks the value of the new `detection_stage` property in `BaseSinkTestWithoutTelemetry`:
- If `DetectionStage.REQUEST` (the default value), `test_insecure` passes the request as always. 
- If `DetectionStage.STARTUP` it passes a `None` request.

NOTE: Another option could be to add a `request` argument to the `test_insecure` method and pass a `None` request when necessary. 

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/build/docker/nodejs/express4/iast/index.js [hunk 1]
```diff
@@ -292,6 +292,16 @@ function initRoutes (app, tracer) {
     searchLdap(filter, req, res)
   })
 
+  app.get('/iast/hardcoded_secrets/test_insecure', (req, res) => {
```
Comment (cbeauchesne) [inline utils/build/docker/nodejs/express4/iast/index.js:295]: Can you add on docs/weblog a section explaining what this endpoint should be doing ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Can you also add an entry in `manifests/cpp.yml` ?

**Review by cbeauchesne — APPROVED** [review]
Failures in CI are not related

File: manifests/nodejs.yml [hunk 2]
```diff
@@ -20,6 +20,10 @@ tests/:
           TestCommandInjection:
             '*': missing_feature
             express4: v3.11.0
+        test_hardcoded_secrets.py:
+          Test_HardcodedSecrets:
+            '*': missing_feature
+            express4: v4.18.0
```
Comment (simon-id) [inline manifests/nodejs.yml:26]: is it not compatible with typescript and next ?

File: tests/appsec/iast/sink/test_hardcoded_secrets.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    location_map = {"nodejs": "iast/index.js"}
+    detection_stage = DetectionStage.STARTUP
+
+    @missing_feature(library="nodejs", reason="Not implemented yet")
+    def test_telemetry_metric_instrumented_sink(self):
+        super().test_telemetry_metric_instrumented_sink()
+
+    @missing_feature(library="nodejs", reason="Not implemented yet")
+    def test_telemetry_metric_executed_sink(self):
+        super().test_telemetry_metric_executed_sink()
```
Comment (simon-id) [inline tests/appsec/iast/sink/test_hardcoded_secrets.py:28]: why is this not implemented yet ? :thinking:

Comment (iunanua) [reply tests/appsec/iast/sink/test_hardcoded_secrets.py:28]: skipped at the moment

File: manifests/nodejs.yml (hunk 2 above)
Comment (iunanua) [reply manifests/nodejs.yml:26]: yes, but we are not testing every iast capability in all frameworks

should next and typescript included @uurien?

Comment (uurien) [reply manifests/nodejs.yml:26]: we only have implemented iast features in express4. 
We should add the tests in other apps also, but I think that it is not the goal of this PR.

**Review by simon-id — APPROVED** [review]

**Review by uurien — APPROVED** [review]

**Pull request merged**